src/Raven.Server/Documents/Sharding/ShardedContext.cs

```diff
@@ -14,11 +16,11 @@ public unsafe class ShardedContext
     {
         public const int NumberOfShards = 1024 * 1024;
 
-        private readonly DatabaseRecord _record;
+        private DatabaseRecord _record;
         public RequestExecutor[] RequestExecutors;
         private readonly long _lastClientConfigurationIndex;
 
-        public ShardedContext(ServerStore server, DatabaseRecord record)
+        public ShardedContext(ServerStore server, RawDatabaseRecord record)
         {
             //TODO: reduce the record to the needed fields
             _record = record;
```

Comment:
you are converting that Raw to materialized anyway here

Reply:
I know... this needs to be handled properly and take only relevant fields. 
change the `TODO` to `TODO: [sharding]`